protocols/horizon/main.go

```diff
@@ -518,7 +518,7 @@ type Transaction struct {
 	OperationCount    int32     `json:"operation_count"`
 	EnvelopeXdr       string    `json:"envelope_xdr"`
 	ResultXdr         string    `json:"result_xdr"`
-	ResultMetaXdr     string    `json:"result_meta_xdr"`
+	ResultMetaXdr     *string   `json:"result_meta_xdr"`
```

Comment:
to meet the requirements for null, need a pointer here can't use empty string.

Reply:
could you use an empty string with the omitempty tag?

```suggestion
	ResultMetaXdr     string    `json:"result_meta_xdr,omitempty"`
```

Reply:
yes, good suggestion, the absence of the field vs. the explicit null of the field has some semantic difference in json parsing, but for the requirements here of 'null', I'm not sure if maybe the absence of field could suffice for same meaning or be too strong? 

There seems to be precedence for omission, i.e. absence of other fields in here, so seems like could follow this pattern for `result_meta_xdr` and avoid the extra noise of compile time breaking change in the go client sdk, it would entail an empty string be present in the `Transaction.ResultMetaXdr` rather than nil, which is more just go/json aspect. I think given that, will move forward with using the `omitempty` and absence of field.

on the wire, the absence of the field is logically equivalent to the field being present and set to null, and still equates to an api breaking change for being an optional field in json parsing terms for any other sdk/client.

Reply:
@mollykarcher , do you see any concern with the criteria of null on meta for  https://github.com/stellar/go/issues/5221 if we make the api json tx meta field `result_meta_xdr` be absent instead of present but set to `result_meta_xdr:null`? as outlined in comments here, there's a slight nuance in the go client sdk for empty string representation in that case, but does avoid an overall breaking compile time change on the go client sdk.

Reply:
I think that's alright. It gets at the same intent of, clients expecting it to be there and trying to grab it by key or something would break.